gateway/src/main/java/com/crud/gateway/service/StudentService.java

```diff
@@ -0,0 +1,84 @@
+package com.crud.gateway.service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.crud.gateway.dto.StudentDto;
+import com.crud.gateway.dto.StudentInputDto;
+import com.crud.gateway.entity.Student;
+import com.crud.gateway.exception.StudentNotFoundException;
+import com.crud.gateway.repository.StudentRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class StudentService {
+
+    private final StudentRepository studentRepository;
+    private final NamesService namesService;
+
+    public List<StudentDto> getAllStudents() {
+        return studentRepository.findAll()
+            .stream()
+            .map((Student s) -> {
+                StudentDto names = this.namesService.getNames(s.getId());
+                return this.buildStudentDto(s.getId(), s.getFirstName(), names.getLastName(), names.getMiddleName());
+            })
+            .collect(Collectors.toList());
+    }
+
+    public StudentDto getStudentById(UUID id) {
+        Student student = studentRepository.findById(id)
+            .orElseThrow(StudentNotFoundException::new);
+
+        StudentDto names = this.namesService.getNames(id);
+
+        return this.buildStudentDto(student.getId(), student.getFirstName(), names.getLastName(),
+                names.getMiddleName());
+    }
+
+    public StudentDto createStudent(StudentInputDto input) {
+        Student student = new Student();
+        student.setFirstName(input.getFirstName());
+
+        studentRepository.save(student);
```

Comment:
`studentRepository.save(toStudent(studentDto.getFirstName()));`
please create new transformer for Student entity.